### PR TITLE
fixed changeMode issue with Profiles

### PR DESCRIPTION
**Pull request by thibmonsel**
This function add mandatory attribute mode for PT.
Fixed changeMode bug with Profiles object.

### CODE REVIEW

File: src/types/PerformanceTable.cpp [hunk 1]
```diff
@@ -395,4 +427,23 @@ void PerformanceTable::display() {
...
     }
   }
-}
\ No newline at end of file
+}
+
+bool PerformanceTable::operator==(const PerformanceTable &pt) const {
+
+  if (this->pt_.size() != pt.getPerformanceTable().size() &&
+      this->pt_[0].size() != pt.getPerformanceTable().size()) {
```
Comment (Mostah) [inline src/types/PerformanceTable.cpp]: shouldn't the check here be `pt_.size != pt.size || pt_[0].size != pt[0].size` ?

File: test/learning/TestInitializeProfile.cpp [hunk 2]
```diff
@@ -175,8 +175,16 @@ TEST(TestProfileInitializer, TestInitializeProfiles) {
   ProfileInitializer profInit = ProfileInitializer(conf, alt_perf);
   MRSortModel model = MRSortModel(3, 4);
   profInit.initializeProfiles(model);
+  std::cout << model.profiles << std::endl;
```
Comment (Mostah) [inline test/learning/TestInitializeProfile.cpp]: You probably forgot to remove the `std::cout` from your test files

File: test/types/TestChangeMode.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#include "../../include/types/PerformanceTable.h"
+#include "../../include/utils.h"
+
+#include "gtest/gtest.h"
+#include <sstream>
+#include <utility>
+
+// Checking PerformanceTable changeModes
+
+TEST(TestPerformanceTable, TestChangeMode1) {
```
Comment (Mostah) [inline test/types/TestChangeMode.cpp]: If you don't execute any test in here, best remove it :)

File: src/types/PerformanceTable.cpp (hunk 1 above)
Comment (thibmonsel) [reply src/types/PerformanceTable.cpp]: you have to call getPerformanceTable() for pt.

File: test/types/TestChangeMode.cpp (hunk 3 above)
Comment (thibmonsel) [reply test/types/TestChangeMode.cpp]: Yup didn't see I added it.

File: src/types/PerformanceTable.cpp (hunk 1 above)
Comment (Mostah) [reply src/types/PerformanceTable.cpp]: The comment was about the logic (&& -> ||) and I believe you forgot a [0] in the second part

Comment (thibmonsel) [reply src/types/PerformanceTable.cpp]: yup thanks for that I got tired probably ahah.

**Review by Mostah — APPROVED** [review]

**Pull request merged**